src/rest-server/src/models/v2/virtual-cluster/k8s.js

```diff
@@ -144,90 +147,90 @@ const getNodeResource = async () => {
-      countedJob.add(pod.userName + '~' + pod.jobName);
-      vcInfos[pod.virtualCluster].numJobs += 1;
-    }
-    mergeDict(vcInfos[pod.virtualCluster].resourcesUsed, pod.resourcesUsed, (x, y) => x + y);
-  }
+  Object.keys(virtualCellCapacity)
+    .concat(['default'])
+    .forEach((vc) => {
+      vcInfos[vc] = {
+        capacity: 0,
+        usedCapacity: 0,
+        dedicated: false,
+        resourcesUsed: {...resourcesEmpty},
+        resourcesGuaranteed: {...resourcesEmpty},
+        resourcesTotal: {...resourcesEmpty},
+      };
+    });
 
-  // set configured resource
+  // set resources
   if (launcherConfig.enabledHived) {
-    const availableTypes = {};
-    for (let vc of Object.keys(virtualCellCapacity)) {
-      availableTypes[vc] = JSON.parse(JSON.stringify(virtualCellCapacity[vc].limit));
-      vcInfos[vc].resourcesTotal = {
-        cpu: Object.values(virtualCellCapacity[vc].quota).reduce((sum, resources) => sum + resources.cpu, 0),
-        memory: Object.values(virtualCellCapacity[vc].quota).reduce((sum, resources) => sum + resources.memory, 0),
-        gpu: Object.values(virtualCellCapacity[vc].quota).reduce((sum, resources) => sum + resources.gpu, 0),
-      };
-    }
-    // minus resources in preempted nodes
-    const preemptedNodes = await fetchNodes(false);
-    for (let node of preemptedNodes) {
-      if (!(node.metadata.name in clusterNodeGpu)) {
-        continue;
-      }
-      const bindings = clusterNodeGpu[node.metadata.name].bindings;
-      for (let vc of Object.keys(bindings)) {
-        if (vc in availableTypes && bindings[vc].type in availableTypes[vc]) {
-          mergeDict(availableTypes[vc][bindings[vc].type], bindings[vc], (x, y) => x - y);
-        }
+    let vcStatus;
+    try {
+      vcStatus = (await axios.get(`${launcherConfig.hivedWebserviceUri}/v1/inspect/clusterstatus/virtualclusters/`)).data;
+    } catch (error) {
+      if (error.response != null) {
+        throw createError(
+          error.response.status,
+          'UnknownError',
+          error.response.data || 'Hived scheduler cannot inspect virtual clusters.',
+        );
+      } else {
+        throw error;
       }
     }
-    // minus used resources in other virtual clusters
-    for (let pod of pods) {
-      const bindings = clusterNodeGpu[pod.nodeIp].bindings;
-      for (let vc of Object.keys(bindings)) {
-        if (vc !== pod.virtualCluster && vc in availableTypes && bindings[vc].type in availableTypes[vc]) {
-          mergeDict(availableTypes[vc][bindings[vc].type], pod.resourcesUsed, (x, y) => x - y);
+    // used, guaranteed, total resources
+    for (let vc of Object.keys(virtualCellCapacity)) {
+      if (vc in vcStatus) {
+        const cellQueue = [...vcStatus[vc]];
+        while (cellQueue.length > 0) {
+          const curr = cellQueue.shift();
+          if (curr.cellPriority === -1) {
+            continue;
+          }
+          if (curr.children) {
```

Comment:
Fully trust hived result. Do not rewrite fields.
But this will depend on new version Hived, v0.3.1

Reply:
updated